CHANGELOG.md

```diff
@@ -14,6 +14,20 @@ go install github.com/riverqueue/river/cmd/river@latest
 river migrate-up --database-url "$DATABASE_URL"
 ```
 
+The migration **includes a new index**. Users with a very large job table may want to consider raising the index separately using `CONCURRENTLY` (which must be run outside of a transaction), then run `river migrate-up` to finalize the process (it will tolerate an index that already exists):
+
+```sql
+ALTER TABLE river_job
+    ADD COLUMN unique_key bytea;
+
+CREATE UNIQUE INDEX CONCURRENTLY river_job_kind_unique_key_idx ON river_job (kind, unique_key) WHERE unique_key IS NOT NULL;
+```
```

Comment:
Dang it, I'm not sure we have a choice except this for the time being. The migration framework automatically uses transactions so that in case a statement within a migration fails, the whole thing can be rolled back without tainting the database.

We'd have to add some new kind of "no transaction" mode to the migrator. Definitely possible, but would take a little more work right now.

Reply:
Might be a silly idea, but I felt the need to throw it out there:

inside the migration you could detect if the index was not created manually and the jobs table is above a certain (unfortunately arbitrary) threshold of rows, and fail the migration with an error indicating that the index should be created concurrently manually?

Reply:
@brandur btw, creating a new type of migration job that runs without a tx would also not help with users who export sql to run in their own migration frameworks.

Reply:
Yeah that's a good thought.

It's still a little difficult because given a very large table, even doing a `SELECT count(*)` can sometimes be a bit of a debacle. Sill though, cheaper than building an index.